### DISCUSSION TITLE
Interactive image zooming and intensity profile plotting

### DISCUSSION THREAD

**Discussion by nghia-vo** [zauberzeug/nicegui#3380] [Show and tell]
I've just added new features to [broh5 software](https://github.com/algotom/broh5)  that allow users to display a zoomed area or the intensity profile of an image by clicking with the mouse. There're [tips](https://github.com/algotom/broh5/blob/f7dfe6e72859e40c2ec529eb1dd5ab0d62641c12/broh5/lib/interactions.py#L121) that can be useful:
- How to get the correct coordinates of a mouse-clicked location when users click on an image displayed by ui.matplotlib 
- How to draw a line or box around the mouse-clicked location using ui.matplotlib.
![broh5_gui](https://github.com/algotom/broh5/raw/main/figs/fig_04.png)

**Comment by Walidbenmoussa** [comment]
I have same problème. I want get x and y in matplotlib mouse move

**Reply by nghia-vo** [reply]
You can bind the event "mousemove" instead of "click":
https://github.com/zauberzeug/nicegui/blob/c3ce92a39688b7f71a064b443e7c87e539281d52/nicegui/elements/echart.js#L18

Example: 
Replace this:
https://github.com/algotom/broh5/blob/f7dfe6e72859e40c2ec529eb1dd5ab0d62641c12/broh5/lib/interactions.py#L86
by:
```py
self.main_plot.on("mousemove", self.mouse_handler)
```

**Comment by moi90** [comment]
@nghia-vo Does that even work if the matplotlib svg element has been resized? (See https://github.com/zauberzeug/nicegui/pull/4595#issuecomment-2787692564)

**Reply by evnchn** [reply]
It doesn't. 

<img src="https://github.com/user-attachments/assets/257350d6-dd0c-4312-ae5f-d7a32aca35f0" width="300">

<img src="https://github.com/user-attachments/assets/b31ed4ab-7b61-4286-b304-9173bd4be298" width="300">

Is because, when you do the profile, then the matplotlib SVG element becomes smaller (in the old version it would involve scrolling), and so the coordinates are offset and incorrect. 

@nghia-vo you should ideally pin NiceGUI version and expect whoever to install `broh5` to do it in a venv, so that upstream NiceGUI changes never affect you like this. Regardless, the fix now is to downgrade NiceGUI with use of `broh5`
